docs/guides/10-integrating-react.md

```diff
@@ -0,0 +1,111 @@
+}
+```
+
+The problems come when it comes to managing the state, either listening to events or synchronizing the attributes.
+
+## Option 1: Manage component logic in the parent layer.
+
+The simplest dirty-hack is to update all attributes, and re-render everything. Even so,  [content attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#Content_versus_IDL_attributes) only accept strings as parameters forcing to set the element properties from javascript.
+
+```jsx
+render() {
+  // We can pass some properties as attributes
+  <as-range-slider id="range" min-value={this.state.minValue} max-value={this.state.maxValue}></as-range-slider>
+}
+
+componentDidMount(){
+  // But we need to use javascript to set callbacks
+  const rangeSlider = document.querySelector('#range');
+  rangeSlider.addEventListener('change', this.onRangeChange.bind(this));
+}
+
+onRangeChange(event){
+  console.log(event.detail);
+}
+```
+
+
+## Option 2: Wrap the component into a react element
+
+With this option we are going to create a React component that wraps the original web-component exposing a simple and react-friendly API.
+
+
+> See the full example [here](https://github.com/CartoDB/airship-demos/tree/master/react/widget)
+
+
+First we create a new [React Component]() to represent our widget, in this case a `category-widget`. In the `render` function we will return
+the airship category widget referenced using a [ref](https://reactjs.org/docs/refs-and-the-dom.html).
+
+
+```js
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef();
+  }
+
+  render() {
+    <as-category-widget ref={this.ref} heading={this.props.heading} description={this.props.description}/>
+  }
+}
+```
+
+Then we will list all availiable properties using [React proptypes](https://reactjs.org/docs/typechecking-with-proptypes.html), this tells react
+which attributes to accept.
+
+```js
+export default class App extends Component {
+   static defaultProps = {
+    categories: [],
+    showHeader: true,
+    showClearButton: true,
+    useTotalPercentage: false,
+    visibleCategories: Infinity,
+  }
+}
+```
+
+Then we set up all the bindings in the `componentDidMount` function binding every React property to the corresponding in the web component. We also associate the events, adding a native listener that connects to `_onSelectedChanged` function.
+
+```js
+export default class App extends Component {
+  componentDidMount() {
+    const widget = ref.current;
+
+    widget.showHeader = this.props.showHeader;
+    widget.showClearButton = this.props.showClearButton;
+    widget.useTotalPercentage = this.props.useTotalPercentage;
+    widget.visibleCategories = this.props.visibleCategories;
+    widget.categories = this.props.categories;
+    widget.addEventListener('categoriesSelected', this._onSelectedChanged.bind(this));
+  }
+
+  _onSelectedChanged(event) {
+    const { onSelectedChanged } = this.props;
```

Comment:
`onSelectedChanged` is not explained.

I get that's a callback prop but it's not listed in propTypes. It's not explained either in the text
`...native listener that connects to '_onSelectedChanged' function. This function will call back the right prop wadus wadus...` You get the point.